contracts/misc/RewardSplitter.sol

```diff
@@ -134,10 +152,49 @@ contract RewardSplitter is IRewardSplitter, Initializable, OwnableUpgradeable, M
     uint256 rewards,
     address receiver
   ) external override returns (uint256 positionTicket) {
+    if (rewards == type(uint256).max) {
+      rewards = rewardsOf(msg.sender);
+    }
     _withdrawRewards(msg.sender, rewards);
     return IVaultEnterExit(vault).enterExitQueue(rewards, receiver);
   }
 
+  function enterExitQueueOnBehalf(uint256 rewards, address onBehalf) external {
+    if (!isClaimOnBehalfEnabled) return;
+
+    if (rewards == type(uint256).max) {
+      rewards = rewardsOf(onBehalf);
+    }
+    _withdrawRewards(onBehalf, rewards);
+    uint256 positionTicket = IVaultEnterExit(vault).enterExitQueue(rewards, address(this));
+    _exitPositions[positionTicket] = onBehalf;
+  }
+
+  function claimExitedAssetsOnBehalf(
+    uint256 positionTicket,
+    uint256 timestamp,
+    uint256 exitQueueIndex
+  ) external {
+    if (!isClaimOnBehalfEnabled) revert Errors.AccessDenied();
+    address onBehalf = _exitPositions[positionTicket];
+    if (onBehalf == address(0)) revert Errors.AccessDenied();
+
+    // calculate exited tickets and assets
+    (uint256 leftTickets, , uint256 exitedAssets) = IVaultEnterExit(vault).calculateExitedAssets(
+      address(this),
+      positionTicket,
+      timestamp,
+      exitQueueIndex
+    );
+    // disallow partial claims (1 ticket could be a rounding error)
+    if (leftTickets > 1) revert Errors.AccessDenied();
+
+    IVaultEnterExit(vault).claimExitedAssets(positionTicket, timestamp, exitQueueIndex);
+    _exitPositions[positionTicket] = address(0);
+
+    Address.sendValue(payable(onBehalf), exitedAssets);
```

Comment:
For gnosis you would need to send ERC-20 tokens, not `sendValue`. I think we should create two version of the RewardSplitter (for gnosis and for eth).